adafruit_atecc/adafruit_atecc.py

```diff
@@ -129,48 +129,29 @@ def __init__(self, i2c_bus, address=_REG_ATECC_DEVICE_ADDR, debug=False):
         """
         self._debug = debug
         self._i2cbuf = bytearray(12)
-        self._i2c_bus = i2c_bus
-        self._i2c_device = None
-        self.wakeup()
-        if not self._i2c_device:
-            self._i2c_device = I2CDevice(self._i2c_bus, address)
-        self.idle()
+        # don't probe, the device will NACK until woken up
+        self._wake_device = I2CDevice(i2c_bus, 0x00, probe=False)
```

Comment:
Is there any alternative to creating two I2CDevice objects or memory management for teardown of _wake_device? The `_wake_device` is only used within calls to `wakeup`.

Reply:
It could maybe be created on the fly within the `wakeup` function. But this would require stashing `i2c_bus` somewhere. Currently, it's only available within `__init__`.

The other alternative would be to use `i2c_bus` directly. But would also require stashing it. And would end up with a bunch of `try_lock()` etc. code. So using an `I2CDevice` seemed cleaner.

Reply:
Ok!